src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,130 @@
+    private final StorageDaoImpl storageDao = new StorageDaoImpl();
+    private User user;
+
+    @BeforeEach
+    void setUp() {
+        user = new User();
+        user.setAge(DEFAULT_AGE);
+        user.setId(DEFAULT_ID);
+        user.setLogin(DEFAULT_LOGIN);
+        user.setPassword(DEFAULT_PASSWORD);
+        Storage.people.clear();
+    }
+
+    @Test
+    void register_checkUser_ok() {
+        registrationService.register(user);
+        User actual = storageDao.get(user.getLogin());
+        assertEquals(actual, user);
+    }
+
+    @Test
+    void register_nullUser_notOk() {
+        user = null;
+        InvalidDataException invalidDataException = assertThrows(InvalidDataException.class,
+                () -> registrationService.register(user));
+        assertEquals(invalidDataException.getMessage(),
+                "User cannot be NULL!");
+    }
+
+    @Test
+    void register_loginAlreadyExists_notOk() {
+        registrationService.register(user);
+        User userCopy = user;
+        InvalidDataException invalidDataException = assertThrows(InvalidDataException.class,
+                () -> registrationService.register(userCopy));
+        assertEquals(invalidDataException.getMessage(),
+                "Login is already taken!");
+    }
+
+    @Test
+    void register_nullLogin_notOk() {
+        user.setLogin(null);
+        InvalidDataException invalidDataException = assertThrows(InvalidDataException.class,
+                () -> registrationService.register(user));
+        assertEquals(invalidDataException.getMessage(),
+                "Login should not be NULL!");
+    }
+
+    @Test
+    void register_loginLength_notOk() {
+        user.setLogin("login");
+        InvalidDataException invalidDataException = assertThrows(InvalidDataException.class, 
+                () -> registrationService.register(user));
+        assertEquals(invalidDataException.getMessage(),
+                "Login must be at least " + MIN_LOGIN_LENGTH
+                + " characters, but was " + user.getLogin().length() + "!");
+    }
+
+    @Test
+    void register_nullPassword_notOk() {
+        user.setPassword(null);
+        InvalidDataException invalidDataException = assertThrows(InvalidDataException.class, 
+                () -> registrationService.register(user));
+        assertEquals(invalidDataException.getMessage(),
+                "The password should not be NULL!");
+    }
+
+    @Test
+    void register_userPasswordLength_notOk() {
+        user.setPassword("12345");
+        InvalidDataException invalidDataException = assertThrows(InvalidDataException.class,
+                () -> registrationService.register(user));
+        assertEquals(invalidDataException.getMessage(),
+                "The password must be at least " + MIN_PASSWORD_LENGTH
+                + " characters, but was " + user.getPassword().length() + "!");
+    }
+
+    @Test
+    void register_notValidUserAge_notOk() {
+        user.setAge(17);
```

Comment:
same here, and probably in some other places too, so please check for the values that could be constants